overlord/managers_test.go

```diff
@@ -362,6 +363,45 @@ func (s *baseMgrsSuite) SetUpTest(c *C) {
 	c.Assert(assertstate.Add(st, a9), IsNil)
 	c.Assert(s.storeSigning.Add(a9), IsNil)
 
+	// add core18 snap declaration
+	headers = map[string]interface{}{
+		"series":       "16",
+		"snap-name":    "core18",
+		"publisher-id": "can0nical",
+		"timestamp":    time.Now().Format(time.RFC3339),
+	}
+	headers["snap-id"] = fakeSnapID(headers["snap-name"].(string))
+	a10, err := s.storeSigning.Sign(asserts.SnapDeclarationType, headers, nil, "")
+	c.Assert(err, IsNil)
+	c.Assert(assertstate.Add(st, a10), IsNil)
+	c.Assert(s.storeSigning.Add(a10), IsNil)
```

Comment:
This might be a silly question, but why are these assertions here, instead of being in a separate test function? Shouldn't the `SetUpTest()` method just be used to setup the test objects and mocks?

Reply:
AFAIU this is just a convenience. The managers tests are kind of nasty to set up correctly. You usually need at least snap declaration, and on occasion snap revision. It takes a bit to figure out how to make each in a correct way.